include/nbl/builtin/hlsl/random/xoroshiro.hlsl

```diff
@@ -0,0 +1,77 @@
+
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_GLSL_RANDOM_XOROSHIRO_HLSL_INCLUDED_
+#define _NBL_BUILTIN_GLSL_RANDOM_XOROSHIRO_HLSL_INCLUDED_
+
+//#include <nbl/builtin/hlsl/math/functions.hlsl>
+#include <nbl/builtin/hlsl/cpp_compat/vector.hlsl>
+
+// TODO[Przemek]: include functions.hlsl instead
+uint32_t rotl(NBL_CONST_REF_ARG(uint32_t) x, NBL_CONST_REF_ARG(uint32_t) k)
+{
+   return (x<<k) | (x>>(32u-k));
+}
```

Comment:
start a `bit` header https://en.cppreference.com/w/cpp/header/bit

implement `rotl` same as in C++ STL except that there's no `constexpr` qualifier :(

also make `mpl/bit.hlsl` and implement a compile time version with templates and `nbl::hlsl::integral_constant`

we'll add to it later